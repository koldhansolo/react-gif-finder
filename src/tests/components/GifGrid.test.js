import { shallow } from "enzyme";
import React from 'react';
import { GifGrid } from "../../components/GifGrid";

// Con este mook finjo cualquier llamada a este archivo y suponer o controlar la info que va a responder
import { useFetchGifs } from '../../hooks/useFetchGifs'; // importo useFetchGifs
jest.mock('../../hooks/useFetchGifs'); // Defino el mock y asocio useFetchGifs para que el mock lo controle

describe('Pruebas a <GifGrid />', () => {

	// Defino una categoria por default
	const category = 'One punch';

	test('Hacer el match con el snapshot', () => {
		// defino el mock para useFetchGifs en el inicio

		// Falseo la data del useFetchGifs
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true
		});

		// Obtengo el wrapper por medio del shallow
		const wrapper = shallow( <GifGrid category={ category } /> );

		// Genero el Snapshot
		expect( wrapper ).toMatchSnapshot();
	});

	test('Verificar que se muestren los items cuando se carguen las imagenes con useFetchGifs', () => {
		// defino un gif de ejemplo
		const gifs = [{
			id: 'ABC',
			url: 'https://media4.giphy.com/media/ZvSojyTPq4OXe/giphy.gif?cid=9c607b57zhg2p5uq15jj5sijh04r5idleqz3u8et76t5lsvd&rid=giphy.gif',
			title: 'cualquier cosa'
		}];
		// defino el mock para useFetchGifs en el inicio

		// Falseo la data del useFetchGifs
		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false
		});

		// Obtengo el wrapper por medio del shallow
		const wrapper = shallow( <GifGrid category={ category } /> );

		// Genero el Snapshot
		expect( wrapper ).toMatchSnapshot();

		expect( wrapper.find('p').exists() ).toBe( false );

		expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
	});
	
});
