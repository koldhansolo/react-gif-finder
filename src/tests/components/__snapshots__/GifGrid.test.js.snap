// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <GifGrid /> Hacer el match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas a <GifGrid /> Verificar que se muestren los items cuando se carguen las imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://media4.giphy.com/media/ZvSojyTPq4OXe/giphy.gif?cid=9c607b57zhg2p5uq15jj5sijh04r5idleqz3u8et76t5lsvd&rid=giphy.gif"
    />
  </div>
</Fragment>
`;
